test_board.py

```diff
@@ -6,26 +6,28 @@ class TestBoard(unittest.TestCase):
     def setUp(self):
         self.test_board = Board()
 
-    def test_has_board(self):
-        expected_result = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
-        result = self.test_board.get_board()
-        self.assertEqual(result, expected_result)
+    def test_board(self):
+        with self.subTest('board exists'):
+            expected_result = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
+            result = self.test_board.get_board()
+            self.assertEqual(result, expected_result)
 
-    def test_board_is_pretty_when_displayed(self):
-        expected_output = "************************\n" \
-                          "*    Current Board!    *\n" \
-                          "************************\n" \
-                          "*                      *\n" \
-                          "*    1  |  2  |  3     *\n" \
-                          "*  ------------------  *\n" \
-                          "*    4  |  5  |  6     *\n" \
-                          "*  ------------------  *\n" \
-                          "*    7  |  8  |  9     *\n" \
-                          "*                      *\n" \
-                          "************************"
-        self.assertEqual(str(self.test_board), expected_output)
+        with self.subTest('board is pretty when displayed'):
+            expected_output = "************************\n" \
+                              "*    Current Board!    *\n" \
+                              "************************\n" \
+                              "*                      *\n" \
+                              "*    1  |  2  |  3     *\n" \
+                              "*  ------------------  *\n" \
+                              "*    4  |  5  |  6     *\n" \
+                              "*  ------------------  *\n" \
+                              "*    7  |  8  |  9     *\n" \
+                              "*                      *\n" \
+                              "************************"
+            self.assertEqual(str(self.test_board), expected_output)
 
-    def test_board_updates_correctly_with_players_move(self):
-        self.test_board.update_board("x", 1)
-        expected_output = [["x", 2, 3], [4, 5, 6], [7, 8, 9]]
-        self.assertEqual(self.test_board.get_board(), expected_output)
+    def test_update_board(self):
+        with self.subTest('board updates correctly with players move'):
```

Comment:
Nice experimentation

Reply:
One thing about test names, you want to be explicit. So for example, what does "update correctly" mean? If I were to look at this test 3 months down the line and saw it fail, I don't know if I would know what it should have been. 
Perhaps another test name that is more explicit is: test_board_updates_cell_specified_by_user_input
So that when test fails with a different value, you know what to look for in the test failure.